multisrc/overrides/heancms/yugenmangas/src/YugenMangas.kt

```diff
@@ -31,6 +42,82 @@ class YugenMangas :
         timeZone = TimeZone.getTimeZone("UTC")
     }
 
+    override fun popularMangaRequest(page: Int): Request {
+        val url = "$apiUrl/query".toHttpUrl().newBuilder()
+            .addQueryParameter("query_string", "")
+            .addQueryParameter("series_status", "All")
+            .addQueryParameter("order", "desc")
+            .addQueryParameter("orderBy", "total_views")
+            .addQueryParameter("series_type", "Comic")
+            .addQueryParameter("page", page.toString())
+            .addQueryParameter("perPage", "12")
+            .addQueryParameter("tags_ids", "[]")
+
+        return GET(url.build(), headers)
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val url = "$apiUrl/query".toHttpUrl().newBuilder()
+            .addQueryParameter("query_string", "")
+            .addQueryParameter("series_status", "All")
+            .addQueryParameter("order", "desc")
+            .addQueryParameter("orderBy", "latest")
+            .addQueryParameter("series_type", "Comic")
+            .addQueryParameter("page", page.toString())
+            .addQueryParameter("perPage", "12")
+            .addQueryParameter("tags_ids", "[]")
+
+        return GET(url.build(), headers)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val sortByFilter = filters.firstInstanceOrNull<SortByFilter>()
+        val statusFilter = filters.firstInstanceOrNull<StatusFilter>()
+
+        val tagIds = filters.firstInstanceOrNull<GenreFilter>()?.state
+            ?.filter(Genre::state)
+            ?.map(Genre::id)
+            .orEmpty()
+            .joinToString(prefix = "[", postfix = "]")
```

Comment:
Set the separator explicitly to `,` as the default includes a space.

```suggestion
        val tagIds = filters.firstInstanceOrNull<GenreFilter>()?.state.orEmpty()
            .filter(Genre::state)
            .map(Genre::id)
            .joinToString(",", prefix = "[", postfix = "]")
```